core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceMigrationMonitorSpec.scala

```diff
@@ -0,0 +1,413 @@
+        } yield {
+          loadedBucket shouldBe defined
+          writeAction
+        }
+      }.unsafeRunSync
+
+      runAndWait(writeAction) shouldBe()
+    }
+  }
+
+  "deleteWorkspaceBucket" should "delete the workspace bucket and record when it was deleted" in {
+    val sourceProject = "general-dev-billing-account"
+    val destBucket = "v1-migration-test-" + UUID.randomUUID.toString.replace("-", "")
+
+    val v1Workspace = minimalTestData.v1Workspace.copy(
+      namespace = sourceProject,
+      googleProjectId = GoogleProjectId(sourceProject),
+      bucketName = destBucket
+    )
+
+    withMinimalTestDatabase { _ =>
+      runAndWait {
+        DBIO.seq(
+          workspaceQuery.createOrUpdate(v1Workspace),
+          WorkspaceMigrationMonitor.schedule(v1Workspace)
+        )
+      }
+
+      val migration = runAndWait(
+        WorkspaceMigrationMonitor.workspaceMigrations
+          .filter(_.workspaceId === v1Workspace.workspaceIdAsUUID).result
+      )
+        .head
+
+      val writeAction = WorkspaceMigrationMonitor.deleteWorkspaceBucket(
+        migration,
+        v1Workspace,
+        new MockGoogleStorageService[IO] {
+          override def deleteBucket(googleProject: GoogleProject, bucketName: GcsBucketName, isRecursive: Boolean, bucketSourceOptions: List[Storage.BucketSourceOption], traceId: Option[TraceId], retryConfig: RetryConfig): fs2.Stream[IO, Boolean] =
+            fs2.Stream.emit(true)
+        }
+      ).unsafeRunSync
+
+      runAndWait(writeAction) shouldBe()
+
+      val workspaceBucketDeleted = runAndWait(
+        WorkspaceMigrationMonitor.workspaceMigrations
+          .filter(_.id === migration.id)
+          .map(_.workspaceBucketDeleted)
+          .result
+          .map(_.head)
+      )
+
+      workspaceBucketDeleted shouldBe defined
+    }
+  }
+
+  "deleteTemporaryBucket" should "delete the temporary bucket and record when it was deleted" in {
+    val sourceProject = "general-dev-billing-account"
+    val sourceBucket = "az-leotest"
+    val destProject = "terra-dev-7af423b8"
+    val destBucket = "v1-migration-test-" + UUID.randomUUID.toString.replace("-", "")
+
+    val v1Workspace = minimalTestData.v1Workspace.copy(
+      namespace = sourceProject,
+      googleProjectId = GoogleProjectId(sourceProject),
+      bucketName = destBucket
+    )
+
+    withMinimalTestDatabase { _ =>
+      runAndWait {
+        DBIO.seq(
+          workspaceQuery.createOrUpdate(v1Workspace),
+          WorkspaceMigrationMonitor.schedule(v1Workspace)
+        )
+      }
+
+      // We need a temp bucket to transfer the workspace bucket contents into
+      val migration = runAndWait(
+        WorkspaceMigrationMonitor.workspaceMigrations
+          .filter(_.workspaceId === v1Workspace.workspaceIdAsUUID).result
+      )
+        .head
+        .copy(
+          newGoogleProjectId = GoogleProjectId(destProject).some,
+          tmpBucketName = GcsBucketName(sourceBucket).some
```

Comment:
This might seem trivial, but can we just use another bucket as the `tmpBucket` here?  I know it technically works for the test but in reviewing it, I had to stop and try to figure out if there was a good reason why it was using `sourceBucket` and it doesn't seem like there is.

```suggestion
          tmpBucketName = GcsBucketName("foo-bar").some
```